src/test/python/parser/opentrackio_lib.py

```diff
@@ -137,60 +234,66 @@ def get_timecode(self):
-        if not part:
-            if self.sample_time_format == "sec":
-                return ssec + (nsec * 0.000000001)
-            elif self.sample_time_format == "timecode":             # since midnight
-                frm = int((nsec * 0.000000001) * self.get_timecode_framerate())
-                return '{:02}'.format(hr) + ":" + '{:02}'.format(mn) + ":" + '{:02}'.format(st) + ":" + '{:02}'.format(frm)
-            elif self.sample_time_format == "string":
-                return "year:{} day:{} hour:{} min:{} sec:{} nsec:{}".format(yr,day,hr,mn,st,nsec)
-        else:
-            if part == 'yy':
-                return yr
-            elif part == 'dd':
-                return day
-            elif part == 'hh':
-                return hr
-            elif part == 'mm':
-                return mn
-            elif part == 'ss':
-                return st
-            elif part == 'ns':
-                return nsec
+        
+            # Calculate the total time as seconds plus fractional seconds from nanoseconds and attoseconds
+            total_time = ssec + (nsec * 1e-9)
+            base_time = datetime(1970, 1, 1) + timedelta(seconds=ssec, microseconds=nsec / 1_000)
+        
+            if not part:
+                t_format = self.sample_time_format
+                if format and format != self.sample_time_format:
+                    t_format = format
+                if t_format == TimeFormat.SECONDS:
+                    return total_time
+                elif t_format == TimeFormat.TIMECODE:
+                    frames = int((nsec * 1e-9) * self.get_timecode_framerate())
+                    return f"{base_time.hour:02}:{base_time.minute:02}:{base_time.second:02}:{frames:02}"
+                elif t_format == TimeFormat.STRING:
+                    return f"year:{base_time.year} day:{base_time.timetuple().tm_yday} hour:{base_time.hour} min:{base_time.minute} sec:{base_time.second} nsec:{nsec}"
+                elif t_format == TimeFormat.ISO8601:
+                    return base_time.strftime(f"%Y-%jT%H:%M:%S.{nsec}")
+            else:
+                # Return individual components based on 'part' argument
+                parts_map = {
+                    'yy': base_time.year,
+                    'dd': base_time.timetuple().tm_yday,
+                    'hh': base_time.hour,
+                    'mm': base_time.minute,
+                    'ss': base_time.second,
+                    'ns': nsec
+                }
+                return parts_map.get(part)
+        
         return None
 
     def get_timecode_framerate(self):
         """Frame rate which the house timecode represents"""
-        if self.validate_dict_elements(self.pd,["timing","sampleRate","num"]):
-            numerator = float(self.pd["timing"]["sampleRate"]["num"])  
-            denominator = float(self.pd["timing"]["sampleRate"]["denom"]) 
+        if self.validate_dict_elements(self.pd,["timing","frameRate","num"]):
```

Comment:
As of today the generated static sample still has this: 
"timing": {
    "sampleRate": {
      "num": 24000,
      "denom": 1001
},
Is this changing to "frameRate" via some PR ? If it doesn't this code will fail.

Reply:
sampleRate is the current key name in 0.9.1